python/pyarrow/src/arrow/python/meson.build

```
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+cython_generated_headers = custom_target(
+    'lib-pyx-headers',
+    input: ['../../../lib.pyx'],
+    output: ['lib.h', 'lib_api.h'],
+    command: [
+        'cython',
+        '--cplus',
+        '@INPUT@',
+        '-o',
+        meson.current_build_dir() / 'lib.cc',
```

```suggestion
        '-o',
        '@OUTPUT0@',
```

and `output: ['lib.cc', 'lib.h', 'lib_api.h']`

You can then pass e.g. `[cython_generated_headers[1], cython_generated_headers[2]]` as the generated dep (where is `cython_generated_headers[0]`, i.e. lib.cc, used?)

---

What I mean is, do you not need lib.cc to be one of the source that get compiled? Do you *just* need the headers?

---

Just the headers

---

I suppose an issue in including the `lib.cc` file is that we also need to install these headers to a particular location; leaving it out of the output makes that simpler I think